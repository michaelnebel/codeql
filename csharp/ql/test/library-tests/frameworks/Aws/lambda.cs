using System.Net;
using System.Collections.Generic;

using Amazon.Lambda.Core;
using Amazon.Lambda.APIGatewayEvents;


namespace LambdaTests {
    public class Functions {
        public APIGatewayProxyResponse Get(APIGatewayHttpApiV2ProxyRequest request, ILambdaContext context) {
            string body = request.Body;  // source
            string cookie = request.Cookies[0];  // source

            string rawpath = request.RawPath;  // source
            string rawquery = request.RawQueryString;  // source
            request.PathParameters.TryGetValue("x", out var pathparameter);  // source

            string header = request.Headers["test"];  // source
            request.Headers.TryGetValue("test", out var header2);  // source


            return new APIGatewayProxyResponse {
                StatusCode = 200
            };
        }
    }
}