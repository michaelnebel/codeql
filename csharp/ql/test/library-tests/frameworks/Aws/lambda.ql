import csharp
import semmle.code.csharp.dataflow.ExternalFlow

query predicate awsRemoteSources(DataFlow::ExprNode node) { sourceNode(node, "remote") }
