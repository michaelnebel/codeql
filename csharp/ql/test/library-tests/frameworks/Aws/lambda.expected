| lambda.cs:11:27:11:38 | access to property Body |
| lambda.cs:12:29:12:43 | access to property Cookies |
| lambda.cs:14:30:14:44 | access to property RawPath |
| lambda.cs:15:31:15:52 | access to property RawQueryString |
| lambda.cs:16:13:16:34 | access to property PathParameters |
| lambda.cs:18:29:18:43 | access to property Headers |
| lambda.cs:19:13:19:27 | access to property Headers |
